lib/engine/game/g_1848/game.rb

```diff
@@ -740,6 +755,80 @@ def revenue_for(route, stops)
           k_sum = stops.count { |rl| rl.hex.tile.label.to_s == 'K' }
           super + K_BONUS[k_sum]
         end
+
+        # recievership
+
+        def close_corporation(corporation, quiet: false)
+          @close_corp_count += 1
+          @player_corp_close_count[corporation.owner] += 1
+
+          # boe gets all the tokens
+          corporation.tokens.select(&:used).each do |token|
+            boe_token = Engine::Token.new(@boe)
+            token.swap!(boe_token)
+            @boe.tokens << boe_token
+          end
+
+          # shareholders compensated
+          per_share = corporation.par_price.price
+          payouts = {}
+          @players.each do |player|
+            next if corporation.president?(player)
+
+            amount = player.num_shares_of(corporation) * per_share
+            next if amount.zero?
+
+            payouts[player] = amount
+            corporation.spend(amount, player, check_cash: false, borrow_from: corporation.owner)
+          end
+
+          if payouts.any?
```

Comment:
unless empty

Reply:
fixed. Btw why is unless empty better than if any?

Reply:
any checks nulls so it's less efficient than empty which is just a size check